io.openems.backend.timedata.aggregatedinflux/src/io/openems/backend/timedata/aggregatedinflux/AggregatedInflux.java

```diff
@@ -0,0 +1,404 @@
+						.ofInstant(Instant.ofEpochMilli(timestamp), measurementEntry.getKey()) //
+						.truncatedTo(DurationUnit.ofDays(1));
+				final var truncatedTimestamp = zonedDateTime.toEpochSecond();
+
+				final var maxPoint = Point //
+						.measurement(measurementEntry.getValue()) //
+						.addTag(OpenemsOEM.INFLUXDB_TAG, String.valueOf(influxEdgeId)) //
+						.time(truncatedTimestamp, WritePrecision.S);
+
+				channelPerType.getOrDefault(ChannelType.MAX, emptyList()).stream() //
+						.forEach(entry -> addEntryToPoint.accept(
+								new AddValuesToPoint(entry.getKey(), entry.getValue(), truncatedTimestamp), maxPoint));
+				this.influxConnector.write(maxPoint, this.writeParametersMaxPoints);
+			}
+		}
+	}
+
+	private Map<ZoneId, String> getDayChangeMeasurements(long timestamp) {
+		final var instant = Instant.ofEpochMilli(timestamp);
+		return this.zoneToMeasurement.entrySet().stream() //
+				.filter(entry -> {
+					final var time = instant.atZone(entry.getKey());
+					return time.getHour() == 23 && time.getMinute() >= 55;
+				}) //
+				.collect(toMap(Entry::getKey, Entry::getValue));
+	}
+
+	private void setAvailableSince(int influxEdgeId, String channelName, long availableSince) {
+		this.influxConnector
+				.write(InfluxConnector.buildUpdateAvailableSincePoint(influxEdgeId, channelName, availableSince));
+		this.availableSinceForEdge.computeIfAbsent(influxEdgeId, key -> new ConcurrentHashMap<>()) //
+				.put(channelName, availableSince);
+	}
+
+	private void checkDataAvailable(//
+			int influxEdgeId, //
+			ZonedDateTime time, //
+			Set<ChannelAddress> channels //
+	) throws OpenemsNamedException {
+		final var errorMessage = this.checkDataAvailableOrErrorMessage(influxEdgeId, time, channels);
+		if (errorMessage == null) {
+			return;
+		}
+		throw new InternalTimedataException(errorMessage);
+	}
+
+	/**
+	 * Checks if the data is available. If true returns null otherwise returns the
+	 * reason why the data is not available.
+	 * 
+	 * @param influxEdgeId the id of the edge
+	 * @param time         the requested time
+	 * @param channels     the requested channels
+	 * @return the reason why no data is available or null if the data is available
+	 */
+	private String checkDataAvailableOrErrorMessage(//
+			int influxEdgeId, //
+			ZonedDateTime time, //
+			Set<ChannelAddress> channels //
+	) {
+		final var edgeChannels = this.availableSinceForEdge.get(influxEdgeId);
+		if (edgeChannels == null) {
+			return null;
+		}
+		final var seconds = time.toEpochSecond();
+		for (var channel : channels) {
+			var availableSince = edgeChannels.get(channel.toString());
+			if (availableSince == null) {
+				return "No availableSince %5d for %sdefined channel %s".formatted(//
+						influxEdgeId, AllowedChannels.isChannelDefined(channel.toString()) ? "" : "un",
+						channel.toString());
+			}
+			// TODO remove
+			// available since is in milliseconds
+			if (availableSince > 999_999_999_999L) {
```

Comment:
could you remove this it was just a mistake by me in my migration script, this can not happen with this implementation